api/views/share.py

```diff
@@ -45,6 +46,12 @@ def get_object(self):
         except InvalidID:
             raise serializers.ValidationError('Invalid ID')
 
+        try:
```

Comment:
Try popping this logic into a [custom permissions class](http://www.django-rest-framework.org/api-guide/permissions/#custom-permissions).

`if hasattr(object, attrname):` is a bit more readable and more safe than catching an `AttributeErrror`.
Anything could raise an `AttributeError`, not just the attribute you're checking for.